include/deal.II/lac/vector.templates.h

```diff
@@ -2048,8 +2045,12 @@ Vector<Number>::allocate()
 {
   // make sure that we don't create a memory leak
   Assert (val == 0, ExcInternalError());
-  val = static_cast<Number *>(_mm_malloc (sizeof(Number)*max_vec_size, 64));
-  Assert (val != 0, ExcOutOfMemory());
+
+  // then allocate memory with the proper alignment requirements of 64 bytes
+  const int ierr = posix_memalign ((void **)&val, 64, sizeof(Number)*max_vec_size);
+
```

Comment:
Yes indeed, the only place that really _needs_ alignment is AlignedVector and even that only in case we compile with AVX (otherwise, the matrix-free applications with explicit vectorization fail). On _MSC_VER, I don't think vectorization is enabled at all, so no problem.
For the rest, in particular the use in `vector.templates.h` and `parallel_vector.templates.h`, it is merely a performance thing: Aligned access is faster than unaligned one, but the compiler does emit correct code in any case.